router/internal/expr/expr.go

```diff
@@ -17,41 +17,39 @@ import (
 // If possible function calls should be avoided in the expressions as they are much more expensive
 // See https://github.com/expr-lang/expr/issues/734
 type RequestRootContext struct {
-	Request RequestContext
+	Request Request `expr:"request"`
 }
 
-// RequestContext is the context for the request object in expressions. Be aware, that only value receiver methods
+// Request is the context for the request object in expressions. Be aware, that only value receiver methods
 // are exported in the expr environment. This is because the expressions are evaluated in a read-only context.
-type RequestContext struct {
-	Auth   RequestAuth
-	Header RequestHeaders
-	URL    RequestURL
+type Request struct {
+	Auth   RequestAuth    `expr:"auth"`
+	URL    RequestURL     `expr:"url"`
+	Header RequestHeaders `expr:"header"`
 }
```

Comment:
I'm fine with the current solution, this would only be another suggestion to avoid the type casting

```suggestion
type accessor interface{
        Get(string) string
}

type Request struct {
	Auth   RequestAuth    `expr:"auth"`
	URL    RequestURL     `expr:"url"`
	Header accessor `expr:"header"`
}
```

Reply:
In your proposal, the user won't be able to anymore to access the headers by map. I'm also fine with your direction. We don't have the need yet to do it and an interface is a good way to ensure the right contract.